### PR TITLE
add: Landing page

### PR DESCRIPTION
**Pull request by saurabhchirde**
Used previously written code to create all the components.

What to review:
[App.js](src/App.js) from lines 14 to 27.
[Login.jsx](src/Components/UI/Modal/Login.jsx) from lines 7 to 38
[NavbarLoginButton.jsx](src/Components/UI/Navigation/NavbarLoginButton/NavbarLoginButton.jsx) from lines 9 to 18
[BodyWrapper](src/Components/UI/Wrapper/BodyWrapper.jsx)
[ModalContext](src/Context/ModalContext.js)
[LandingPage](src/Pages/Landing/LandingPage.jsx)

Preview:
pebble-cart

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
+import Footer from "./Components/UI/Footer/Footer";
+import BodyWrapper from "./Components/UI/Wrapper/BodyWrapper";
 
 const App = () => {
-  return <h1>App</h1>;
+  const { showLoginModal, showSignupModal } = useModal();
+  return (
+    <>
+      {showLoginModal && <Login />}
+      {showSignupModal && <Signup />}
```
Comment (t007rushi) [inline src/App.js]: emphasis on using more ternary operator than &&

File: src/Components/UI/Navigation/MobileNavigationBar.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        <a href="/">
+          <img className="logo" src={logoIcon} alt="logo" />
+        </a>
+        <SearchBar
+          searchWrapper="search-container"
+          micIcon="fas fa-microphone"
+          searchIcon="fas fa-search"
+          placeholder="Search"
+          onChange={() => {}}
+          onIconClick={() => {}}
```
Comment (t007rushi) [inline src/Components/UI/Navigation/MobileNavigationBar.jsx:20]: empty action??

File: src/Context/ModalContext.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React, { createContext, useContext, useState } from "react";
+
+const modalContext = createContext();
```
Comment (t007rushi) [inline src/Context/ModalContext.js]: put up default null if want nothing as it causes production issue

**Review by t007rushi — COMMENTED** [review]
Can look for optimized folder structure

File: src/App.js (hunk 1 above)
Comment (saurabhchirde) [reply src/App.js]: Ok, will do that.

File: src/Components/UI/Navigation/MobileNavigationBar.jsx (hunk 2 above)
Comment (saurabhchirde) [reply src/Components/UI/Navigation/MobileNavigationBar.jsx:20]: Written for future usage.

File: src/Context/ModalContext.js (hunk 3 above)
Comment (saurabhchirde) [reply src/Context/ModalContext.js]: Ok, will look into it.

File: src/Components/UI/Button/BadgeIconButton.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+
+const BadgeIconButton = (props) => {
+  return (
+    <>
```
Comment (tarunsankhla) [inline src/Components/UI/Button/BadgeIconButton.jsx:5]: this was something new I learnt from the PR

File: src/Components/UI/Navigation/DesktopNavigationBar.jsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            badgeClassName="badge-on-icon"
+            badgeValue="1"
+            label="Cart"
+            onClick={() => {}}
+          />
+          <NavbarAvatar
+            avatarWrapper="badge-container"
+            avatarClassName="avatar text-avatar-xsm-round"
+            imgDisplay="hide"
+            src=""
```
Comment (tarunsankhla) [inline src/Components/UI/Navigation/DesktopNavigationBar.jsx:48]: src?

File: src/Components/UI/Navigation/NavbarLoginButton/NavbarLoginButton.jsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import Button from "../../Button/Button";
+import { useModal } from "../../../../Context/ModalContext";
+
+const NavbarLoginButton = () => {
+  const { setShowLoginModal, setShowSignupModal, loginButton, setLoginButton } =
+    useModal();
+
+  const onNavbarLoginClickHandler = () => {
```
Comment (tarunsankhla) [inline src/Components/UI/Navigation/NavbarLoginButton/NavbarLoginButton.jsx:9]: add comments in you code about what action does the  method performs

**Review by tarunsankhla — APPROVED** [review]
overall it is good but try to split the PR to many lines of code to review

File: src/Components/UI/Navigation/DesktopNavigationBar.jsx (hunk 5 above)
Comment (saurabhchirde) [reply src/Components/UI/Navigation/DesktopNavigationBar.jsx:48]: src prop is provided to show the user's profile picture after logging in.

**Review by t007rushi — APPROVED** [review]
nice work

Alt Text

**Pull request merged**